### PR TITLE
update matter lib to 0.10.x and Add AsyncStorage

### PR DESCRIPTION
**Pull request by Apollon77**
This also includes upgrading all clusters and such to Matter 1.3

### CODE REVIEW

File: src/matter/ControllerNode.ts [hunk 1]
```diff
@@ -713,13 +714,18 @@ class Controller {
...
         // this.adapter.log.debug(`Commissioning ... ${JSON.stringify(options)}`);
         try {
-            const nodeObject = await this.commissioningController.commissionNode(options);
-            this.matterNodeIds.push(nodeObject.nodeId);
+            const nodeId = await this.commissioningController.commissionNode(options);
+            this.matterNodeIds.push(nodeId);
+            const nodeObject = this.commissioningController.getConnectedNode(nodeId);
+            if (nodeObject === undefined) {
+                // should never happen
+                throw new Error(`Node ${nodeId} not found connected but commissioning was successful.`);
```
Comment (foxriver76) [inline src/matter/ControllerNode.ts:722]: `not found connected`, maybe `is not connected`?

File: src/matter/IoBrokerNodeStorage.ts [hunk 2]
```diff
@@ -45,66 +39,74 @@ export class IoBrokerNodeStorage implements Storage {
...
 
         this.initialized = true;
     }
 
     async clearAll(): Promise<void> {
-        await this.adapter.delObjectAsync(this.storageRootOid, { recursive: true });
+        try {
+            await this.adapter.delObjectAsync(this.storageRootOid, { recursive: true });
+        } catch (error) {
+            this.adapter.log.error(`[STORAGE] Cannot clear all state: ${error}`);
```
Comment (foxriver76) [inline src/matter/IoBrokerNodeStorage.ts:56]: logging error object on purpose or is `error.message` enough? When being explicit maybe `error.stack`?

File: src/matter/IoBrokerNodeStorage.ts [hunk 3]
```diff
@@ -45,66 +39,74 @@ export class IoBrokerNodeStorage implements Storage {
...
-        const value = this.data[this.buildKey(contexts, key)];
-        if (value === null || value === undefined) {
-            return undefined;
+        try {
+            const valueState = await this.adapter.getStateAsync(this.buildKey(contexts, key));
+            if (valueState === null || valueState === undefined) {
+                return undefined;
+            }
+            if (typeof valueState.val !== 'string') {
+                this.adapter.log.error(`[STORAGE] Invalid value for key ${key} in context ${contexts.join('$$')}: ${toJson(valueState.val)}`);
```
Comment (foxriver76) [inline src/matter/IoBrokerNodeStorage.ts:80]: maybe put key and context in `"`

File: src/matter/IoBrokerNodeStorage.ts [hunk 4]
```diff
@@ -45,66 +39,74 @@ export class IoBrokerNodeStorage implements Storage {
...
+                return undefined;
+            }
+            return fromJson(valueState.val) as T;
+        } catch (error) {
+            this.adapter.log.error(`[STORAGE] Cannot read state: ${error}`);
```
Comment (foxriver76) [inline src/matter/IoBrokerNodeStorage.ts:85]: same with error object

File: src/matter/IoBrokerNodeStorage.ts [hunk 5]
```diff
@@ -194,8 +164,12 @@ export class IoBrokerNodeStorage implements Storage {
             throw new StorageError('[STORAGE] Context and key must not be empty strings!');
         }
         const oid = this.buildKey(contexts, key);
-        delete this.data[oid];
 
-        this.deleteKey(oid);
+        try {
+            await this.adapter.delObjectAsync(oid);
+        } catch (error) {
+            this.adapter.log.error(`[STORAGE] Cannot delete state: ${error}`);
```
Comment (foxriver76) [inline src/matter/IoBrokerNodeStorage.ts:171]: same with error obj

**Review by foxriver76 — APPROVED** [review]
LGTM just some small comments

File: src/matter/IoBrokerNodeStorage.ts (hunk 2 above)
Comment (GermanBluefox) [reply src/matter/IoBrokerNodeStorage.ts:56]: This error is so seldom and can only occur if no permissions or object not exist. So it is definitely enough just to indicate `error.toString()` to understand, what happens

**Review by GermanBluefox — APPROVED** [review]
LGTM. imho

File: src/matter/IoBrokerNodeStorage.ts (hunk 2 above)
Comment (Apollon77) [reply src/matter/IoBrokerNodeStorage.ts:56]: i see the same. Adjusted

**Pull request merged**